core/Pos/Aeson/Core.hs

```diff
@@ -39,14 +50,43 @@ deriving instance ToJSON SlotCount
 instance FromJSON CoinPortion where
     parseJSON v = unsafeCoinPortionFromDouble <$> parseJSON v
 
+instance ToJSON CoinPortion where
+    toJSON = toJSON . coinPortionToDouble
+
+instance FromJSON Address => FromJSONKey Address where
+    fromJSONKey = FromJSONKeyTextParser (eitherToFail . decodeTextAddress)
+
+instance ToJSONKey Address where
+    toJSONKey = toJSONKeyText (sformat addressF)
+
 deriveFromJSON S.defaultOptions ''VssCertificate
 deriveFromJSON S.defaultOptions ''Millisecond
 deriveFromJSON S.defaultOptions ''Microsecond
 deriveFromJSON S.defaultOptions ''Second
 deriveFromJSON S.defaultOptions ''SoftforkRule
 deriveFromJSON S.defaultOptions ''BlockVersionData
+deriveToJSON   S.defaultOptions ''Microsecond
 
 deriving instance FromJSON Timestamp
+deriving instance ToJSON Timestamp
+
+-- Instances needed for wallet
+deriveFromJSON defaultOptions ''SlotId
+deriveFromJSON defaultOptions ''BlockCount
+deriveFromJSON defaultOptions ''LocalSlotIndex
+deriveFromJSON defaultOptions ''ChainDifficulty
+
+instance FromJSON UnparsedFields where
+    parseJSON v = UnparsedFields . Map.map getJsonByteString <$> parseJSON v
+
+instance ToJSON UnparsedFields where
+    toJSON (UnparsedFields fields) = toJSON (Map.map JsonByteString fields)
+
+deriveJSON defaultOptions ''Attributes
+deriveJSON defaultOptions ''AddrAttributes
+deriveJSON defaultOptions ''AddrStakeDistribution
+deriveJSON defaultOptions ''AddrType
+deriveJSON defaultOptions ''Address
```

Comment:
It looks strange to format `Address`-as-a-key-in-a-dictionary and separate `Address` differently.